### ISSUE TITLE
Cannot close dropdown with folders containing index.md

### ISSUE THREAD

**Issue by mishushakov** [fuma-nama/fumadocs#1127]
### To Reproduce

file structure:
<img width="278" alt="Screenshot 2024-11-29 at 16 16 30" src="https://github.com/user-attachments/assets/0f73de84-7c77-4021-88e7-1c48fe8fc498">

http://github.com/mishushakov/fuma-docs-repro

### Current vs. Expected behavior

See difference between the two
https://github.com/user-attachments/assets/a9376e94-11a6-4d09-bb49-7afb92730c44

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.6.0: Mon Jul 29 21:14:30 PDT 2024; root:xnu-10063.141.2~1/RELEASE_ARM64_T6000
  Available memory (MB): 16384
  Available CPU cores: 8
Binaries:
  Node: 20.12.2
  npm: 10.5.0
  Yarn: 1.22.21
  pnpm: 9.12.2
Relevant Packages:
  next: 15.0.3 // Latest available version is detected (15.0.3).
  eslint-config-next: 15.0.3
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.6.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

UI

**Comment by fuma-nama** [comment]
addressed on 14.5.5, to be released soon

**Comment by mishushakov** [comment]
Thank you so much!
